attest/wrapped_tpm20.go

```diff
@@ -205,67 +218,65 @@ func (t *wrappedTPM20) getKeyHandleKeyMap() (map[crypto.PublicKey]tpmutil.Handle
 		if pub.RSAParameters != nil || pub.ECCParameters != nil {
 			key, err := pub.Key()
 			if err != nil {
-				return nil, err
+				return nil, nil, err
+			}
+			serializedKey, err := serializePublicKey(key)
+			if err != nil {
+				return nil, nil, err
 			}
-			rvalue[key] = keyHandle
+			key2Handle[serializedKey] = keyHandle
+			handleFound[keyHandle] = struct{}{}
+		}
+	}
+	return key2Handle, handleFound, nil
+}
+
+func (t *wrappedTPM20) create2KRSAEKInAvailableSlot(handleFoundMap map[tpmutil.Handle]struct{}) (tpmutil.Handle, error) {
+	rsakeyHandles := []tpmutil.Handle{
+		commonRSAEkEquivalentHandle,
+		commonECCEkEquivalentHandle + 1,
+		commonECCEkEquivalentHandle + 2,
+	}
+	for _, targetHandle := range rsakeyHandles {
+		_, handleInUse := handleFoundMap[targetHandle]
+		if handleInUse {
+			continue
+		}
+		ekTemplate := t.rsaEkTemplate()
+		err := t.createEK(ekTemplate, targetHandle, fmt.Errorf(""))
+		if err != nil {
+			return tpmutil.Handle(0), err
 		}
+		return targetHandle, nil
 	}
-	return rvalue, nil
+	return tpmutil.Handle(0), fmt.Errorf("could not create rsa 2048 key in persistent handle")
```

Comment:
nit: this can only occur if all the handles are in use, so the error comment can be updated to reflect this. 
"no available handles to create RSA 2048 key in persistent handle"